responsebot/responsebot_client.py

```
@@ -51,9 +51,17 @@ def tweet(self, text, in_reply_to=None):
 
         :param text: the text to post
         :param in_reply_to: The ID of the tweet to reply to
+        :raise TweetError: if the tweet exceeds Twitter's character limit or the tweet is duplicated
         :return: Tweet object
         """
-        return Tweet(self._client.update_status(status=text, in_reply_to_status_id=in_reply_to)._json)
+        if len(text) > TWITTER_CHARACTER_LIMIT:
+            raise TweetError(reason="Message exceeds Twitter's character limit")
+
+        try:
+            return Tweet(self._client.update_status(status=text, in_reply_to_status_id=in_reply_to)._json)
+        except TweepError as e:
+            if e.api_code == TWITTER_DUPLICATED_TWEET_ERROR:
+                raise TweetError(reason='Duplicated tweet')
```

@tklarryonline other errors are assumed to raise errors (& thus handled by our error handling outside of this scope). I only re-raised the errors that are stated to be `fail silently` in their [docs](http://tweepy.readthedocs.io/en/v3.5.0/api.html#status-methods)